### PR TITLE
Configure caBundle in apiservice/v1beta1.custom.metrics.k8s.io by service.alpha.openshift.io/inject-cabundle annotation

### PR DESCRIPTION
**Pull request by nak3**
This patch contains following changes:

- adds `service.alpha.openshift.io/inject-cabundle`
- changes insecureSkipTLSVerify to `false`.

in `apiservice/v1beta1.custom.metrics.k8s.io`.

After this patch `caBundle` is inserted automatically thanks to `service.alpha.openshift.io/inject-cabundle` annotation.
```
$ oc get apiservice v1beta1.custom.metrics.k8s.io -o yaml |grep caBundle
  caBundle: LS0tLS1CRUdJTiBDRVJUSU... snip ...
```

Fixes https://jira.coreos.com/browse/SRVKS-194

cc @maschmid @markusthoemmes

### CODE REVIEW

File: pkg/controller/knativeserving/openshift/openshift.go [hunk 1]
```diff
@@ -194,6 +194,43 @@ func installServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
 	return nil
 }
 
+// removeCaBundleFromApiservice removes spec.caBundle from v1beta1.custom.metrics.k8s.io.
+// This is necessary as `insecureSkipTLSVerify: true` in resources/knative-serving-x.y.z.yaml conflicts
+// with service.alpha.openshift.io/inject-cabundle annotaiton in addCaBundleToApiservice.
+func removeCaBundleFromApiservice(instance *servingv1alpha1.KnativeServing) error {
```
Comment (nak3) [inline pkg/controller/knativeserving/openshift/openshift.go]: I know that this is ugly hack. But applying `resources/knative-serving-0.7.1.yaml` second time (= when operator restarts) produces following error :crying_cat_face:

```
"error":"APIService.apiregistration.k8s.io \"v1beta1.custom.metrics.k8s.io\" is invalid: spec.insecureSkipTLSVerify: Invalid value: true: may not be true if caBundle is present"
```

@jcrossley3  @markusthoemmes @maschmid Please let me know if  you have any better idea.

Comment (markusthoemmes) [reply pkg/controller/knativeserving/openshift/openshift.go]: Can we not transform the existing APIService programatically like we do below with some deployments?

File: pkg/apis/addtoscheme_serving_v1alpha1.go [hunk 2]
```diff
@@ -2,9 +2,10 @@ package apis
 
 import (
 	"github.com/openshift-knative/knative-serving-operator/pkg/apis/serving/v1alpha1"
+	"k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1"
 )
 
 func init() {
 	// Register the types with the Scheme so the components can map objects to GroupVersionKinds and back
-	AddToSchemes = append(AddToSchemes, v1alpha1.SchemeBuilder.AddToScheme)
+	AddToSchemes = append(AddToSchemes, v1alpha1.SchemeBuilder.AddToScheme, v1beta1.SchemeBuilder.AddToScheme)
```
Comment (markusthoemmes) [inline pkg/apis/addtoscheme_serving_v1alpha1.go]: I feel like this should move to the  `Configure` function of the Openshift plugin instead. Is there a similar `Install` function available as we use there to install Openshift APIs?

Comment (nak3) [reply pkg/apis/addtoscheme_serving_v1alpha1.go]: Yes, agree :+1:  Updated.

**Pull request merged**